### ISSUE TITLE
Feature suggestion: Steam-powered pump

### ISSUE THREAD

**Issue by thriem** [Deadlock989/IndustrialRevolution#152]
Reasoning:
Since "bronze-age" and its steam powered eco-system becomes obsolete rather soon in game and yet is one of the outstanding mechanics of the mod, I see the lack of a "steam-powered pump" a missed opportunity to elevate new gameplay mechanics.
Once electric infrastructure becomes available, the desire of replacing everything at once is imminent. So prioritisation takes place. Miners and some structures may be reused for (remote) outposts - if delivery of steam could be made by trains, which now isn't reliable, since you need to kick off electricity for pumps to run - and if the idle-consumption drains all the steam, it needs to be re-ignited again.

Suggested Feature:
Entity of a steam-powered pump.

Potential gains:
- it would enable outposts to be powered by steam-delivery (by trains), thus, effectively enabling it to run without the need for electric-pole trenches - or steam-pipes, if so desired.
- can smoothen the transition from steam- to electricity-powered bases, by said reusability.
- can also allow, to produce electricity on site, without the need of poles - which makes the construction of outposts a bit faster and more independent.

**Comment by Deadlock989** [comment]
This would be extremely difficult.

The lesser problem is that steam-powered machines need a separate fluid connector. I don't know if pumps even support fluid power connections (the docs don't say that they don't, unlike some other entities where it's explicitly ruled out) but assuming they do, you have a 1x2 entity that now needs (at least) a third fluid connection point. Rotations and pipe placement would be very awkward. Having it as a burner entity would be slightly less awkward, but read on.

The real issue is that pump graphics are horribly complex. The vanilla pump has a staggering 110 sprite sheets (sheets, not sprites) for each of the SD and HD resolutions. Mostly this is because of the elaborate way they connect to fluid wagons, making it by far the most complex animation in the vanilla game. Add a third fluid connector into that picture and I already need to lie down in a darkened room for a bit.

In short, it's not a bad idea, but way beyond my pay grade - it would probably need a full week of work just to prototype it and figure out a workflow, let alone polish and finish it. I'm never going to find that kind of time.

**Comment by Cadams111** [comment]
How feasible would it be to have the steam powered pump be powered by the input end of the standard pump and only be able to transport steam? Like it siphons off some steam from what it’s pumping? I realize this wouldn’t fix the 110 sheet issue, buy it would help with the needing more connections issue.

**Comment by Deadlock989** [comment]
That's not possible - crafting/functional fluidboxes are defined completely separately from energy source fluidboxes. You could maybe do that with on_tick scripting, but it would not be good for performance.

So functionally you are looking at pipe connections configured like this (if you want the same pass-through steam as every other steam machine):

![image](https://user-images.githubusercontent.com/3128845/210905817-22730b32-8a2a-4f9a-ad50-de75dfc62767.png)

It would work, but it would be easy to get fluids mixed with a careless pipe placement.

But it's moot - the graphics work is too much to commit to this decade, sorry.

**Comment by thriem** [comment]
Could it be, that the amount of sheets come from the different colours of fluids, which are visible? Theoretically, if the pump would be restricted to fill / empty fluid wagons without any variation caused by fluid variety - would that "drastically" reduce the animations? And in that case, be a i.e. 6x2 entity for ease of use / less risk of fluid mixing? But guess that is increasing workload again.
